lib/analyzer/applications/java.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from "path";
+import { JarFingerprint } from "../types";
+import { AppDepsScanResultWithoutTarget, FilePathToContent } from "./types";
+
+function mapContentToJarFingerprints(input: {
+  [fileName: string]: string;
+}): {
+  [path: string]: JarFingerprint[];
+} {
+  const jarFingerprints: JarFingerprint[] = Object.entries(input).map(
+    ([filePath, digest]) => {
+      return {
+        location: filePath,
+        digest,
+      };
+    },
+  );
+
+  const resultAggregatedByPath = jarFingerprints.reduce(
+    (accumulator, current) => {
+      const location = path.dirname(current.location);
+      accumulator[location] = accumulator[location] || [];
+      accumulator[location].push(current);
+      return accumulator;
+    },
+    {},
+  );
+
+  return resultAggregatedByPath;
+}
+
+export async function jarFilesToScannedProjects(
```

Comment:
nit: Hehe, just noticed that `scanned projects` used to be the old name for `ScanResult`. Just something that might want to rename in the future for consistency